### DISCUSSION TITLE
Auto instrumenting java-micronaut lambda function with ADOT layer

### DISCUSSION THREAD

**Discussion by arjunrc143** [open-telemetry/opentelemetry-lambda#238] [Q&A]
I tried to auto-instrument my java micronaut lambda function using ADOT layer (arn:aws:lambda:eu-central-1:901920570463:layer:aws-otel-java-wrapper-ver-1-2-0:2). 

The documentation I followed is https://aws-otel.github.io/docs/getting-started/lambda/lambda-java 

I want send the traces to my elastic backend. Therefore I added additional configurations(Environment variables) to do so. I also tried adding a collector.yaml and overriding the config environment variable. Here is my code, 

https://github.com/arjunrc143/sam-micronaut-function

But with this I am getting the following error and not sure how to proceed from here. 

"errorMessage: Receiver class io.opentelemetry.extension.aws.AwsConfigurablePropagator does not define or inherit an implementation of the resolved method 'abstract io.opentelemetry.context.propagation.TextMapPropagator getPropagator()' of interface io.opentelemetry.sdk.autoconfigure.spi.ConfigurablePropagatorProvider.
errorType: java.lang.AbstractMethodError"

Can anyone please help me in resolving this. Thanks in advance.

**Comment by tylerbenson** [comment]
Do you still need help here? Feel free to reopen if so.
